week3/week3-homework/Program.cs

```diff
@@ -0,0 +1,142 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+
+
+     var celsiusTemperature = new Temperature(17.00m);
+
+        string MakeSound (IAnimal animal){
+          return $"{animal.Name} says {animal.Sound}";
+        }  
+
+        var Cat = new Animal("Cat","meow meow");
+        var Dog = new Animal("Dog","woof woof");
+        var Cow = new Animal("Cow","maa maa");
+        var DogSound = MakeSound(Dog);
+
+app.Run();
+
+
+
+ 
+
+       
+
+
+
+
+// Temperature Class
+public class Temperature
+{
+    private decimal _temperatureAlert = -273.15m;
+    private decimal _degreesCelsius;
+    
+    public decimal DegreesCelsius
+    {
+    get => _degreesCelsius;
+        private set
+    {
+        if (decimal.Compare(_degreesCelsius, _temperatureAlert) < 0)
+        {
+            throw new Exception("this is so cold you should be careful");

```

Comment:
That seems more like a general warning than an exception. Remember - exceptions are for exceptional cases :-)